### PR TITLE
Sa 3334 mtp import update

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [SA-3334](https://jumpcloud.atlassian.net/browse/SA-3334) - Import/Update-JCMspFromCSV

## What does this solve?

Provides a utility for MSP admins to update and create new MSP orgs from a CSV file. This release allows administrators to create MSP orgs, set /update names and set/ update max user counts.

## Is there anything particularly tricky?

Yes, the automated tests for this set of functions is invoked using the pester module but a different set of rules were introduced if a specific MSP Org environment variable is present.

## How should this be tested?

Each of these steps are accounted for in our unit tests except for the Import-JCMSPFromCSV test which is mocked during automated tests.

Access to a testing MSP org is required. 

**Template Tests:**
Create an import and update template. The import template should only have a name and maxSystemUsers field. The update template should contain all names of orgs/ maxSystemUsers and IDs for each org in your tenant.

**Import Tests**
Create a new line in the import template, specify an org name and maxSystemUser count. Import the org using the `Import-JCMSPFromCSV` function. The new org should be created in your tenant.

**Update Tests**
Re-Create the update template (overwrite existing template), the new template CSV should contain the new MSP org. Change the name of the org, change the maxSystemUsers count and delete all other rows of orgs. 

Update the org by running the `Update-JCMSPFromCSV` function. The org created in the import tests should be updated with a new name and new maxSystemUsers count.

## Screenshots

[SA-3334]: https://jumpcloud.atlassian.net/browse/SA-3334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PowerShell/JumpCloud Module/Public/Utilities/CSV_Import/New-JCMSPImportTemplate.ps1 [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        }
+    }
+
+    process {
+        if ($type -eq 'Import') {
+            $ConfirmUpdateVsNew = 'N'
+        } elseif ($type -eq 'Update') {
+            $ConfirmUpdateVsNew = 'U'
+        } Else {
+            Write-Host "`nDo you want to create an import CSV template for creating new MSP orgs or updating exisitng MSP orgs?"
```
Comment (gweinjc) [inline PowerShell/JumpCloud Module/Public/Utilities/CSV_Import/New-JCMSPImportTemplate.ps1:84]:
```suggestion
            Write-Host "`nDo you want to create an import CSV template for creating new MSP orgs or updating existing MSP orgs?"
```

File: PowerShell/JumpCloud Module/Public/Utilities/CSV_Import/New-JCMSPImportTemplate.ps1 [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            If (!(Get-PSCallStack | Where-Object { $_.Command -match 'Pester' })) {
+                Clear-Host
+            }
+
+            Write-Host $Banner -ForegroundColor Green
+            Write-Host "`n$Heading2`n"
+            Write-Host " $PWD" -ForegroundColor Yellow
+            Write-Host ""
+
+
```
Comment (gweinjc) [inline PowerShell/JumpCloud Module/Public/Utilities/CSV_Import/New-JCMSPImportTemplate.ps1:52]:
```suggestion
TODO: Add checks for lower case input or whitespaces
```

**Review by gweinjc — COMMENTED** [review]
Everything tests good in my MTP Tenant. I was able to create 2 new orgs as well as update them with a new name and maxSystemUsers value.

I do have some nitpicky comments regarding the user prompts. I feel like they aren't necessarily intuitive and are kind of confusing, we also leave things open to bad user input.

I think we should do a PromptForChoice approach to eliminate any kind of user error, minus inputting the file paths.

Here's an example:
```powershell
# PromptForChoice Args
$Title = "Do you want to create an import CSV template for creating new MSP orgs or updating existing MSP orgs?"
$Prompt = "Enter your choice"
$Choices = [System.Management.Automation.Host.ChoiceDescription[]] @("&New MSP Orgs", "&Update Existing MSP Orgs", "&Cancel")
$Default = 1

# Prompt for the choice
$Choice = $host.UI.PromptForChoice($Title, $Prompt, $Choices, $Default)

# Action based on the choice
switch($Choice)
{
    0 { Write-Host "New - Write your code"}
    1 { Write-Host "Update - Write your code"}
    2 { Write-Host "Cancel - Write your code"}
}
```

File: PowerShell/JumpCloud Module/JumpCloud.psd1 [hunk 3]
```diff
@@ -150,7 +151,7 @@ PrivateData = @{
  } # End of PrivateData hashtable
 
 # HelpInfo URI of this module
-HelpInfoURI = 'https://github.com/TheJumpCloud/support/wiki'
+HelpInfoURI = 'file://giom/TheJumpCloud/support/wiki'
```
[the remark this answers is not in the document]
Comment (jworkmanjc) [reply PowerShell/JumpCloud Module/JumpCloud.psd1]: fixed, my auto-formatter messed this up. I've had to sometimes open and close VSCode when that happens, looks like I pushed a bad change.

**Review by kmaranionjc — APPROVED** [review]
Tests worked, PID fixed. Nice job!

**Comment by jworkmanjc** [comment]
Hey all, just re-requesting review here. After moving things around to prompt for user choice I had to change tests a bit. Last review I promise:

New-JCMSPImportTemplate
Screen Shot 2023-06-21 at 10 18 09 AM

Update-JCMSPFromCSV
Screen Shot 2023-06-21 at 10 19 35 AM

Import-JCMSPFromCSV
Screen Shot 2023-06-21 at 10 20 32 AM

**Review by gweinjc — APPROVED** [review]

**Pull request merged**